languagetool-gui-commons/src/main/java/org/languagetool/gui/ConfigurationDialog.java

```diff
@@ -177,7 +180,35 @@ public void actionPerformed(@SuppressWarnings("unused") ActionEvent actionEvent)
     cons.fill = GridBagConstraints.HORIZONTAL;
     Collections.sort(rules, new CategoryComparator());
     DefaultMutableTreeNode rootNode = createTree(rules, false);   //  grammar options
-    configTree[0] = new JTree(getTreeModel(rootNode));
+    configTree[0] = new JTree(getTreeModel(rootNode)){
+		/**
+		 *
+		 */
+		private static final long serialVersionUID = 1L;
+
+		public void setState() {
+			setExpandedState(lastPath, state);
```

Comment:
Please indent your code with 2 spaces so it matches the rest of the file.

Reply:
ok,im using eclipse,is there any formatter to be import for eclipse?which is followed by LT

Reply:
I don't know, but please see here: http://wiki.languagetool.org/code-style